src/likelihoods/categorical.jl

```diff
@@ -1,22 +1,45 @@
 """
-    CategoricalLikelihood(l=softmax)
+    CategoricalLikelihood(l=softmax, bijective=true)::CategoricalLikelihood{bijective,typeof(Link(k))}
 
 Categorical likelihood is to be used if we assume that the 
-uncertainity associated with the data follows a Categorical distribution.
+uncertainty associated with the data follows a [Categorical distribution](https://en.wikipedia.org/wiki/Categorical_distribution) with `n` categories.
 ```math
     p(y|f_1, f_2, \\dots, f_{n-1}) = \\operatorname{Categorical}(y | l(f_1, f_2, \\dots, f_{n-1}, 0))
+    p(y|f_1, f_2, \\dots, f_n) = \\operatorname{Categorical}(y | l(f_1, f_2, \\dots, f_n))
 ```
+Two variants are possible:
+## `bijective=true`
 Given an `AbstractVector` ``[f_1, f_2, ..., f_{n-1}]``, returns a `Categorical` distribution,
 with probabilities given by ``l(f_1, f_2, ..., f_{n-1}, 0)``.
+It is used by default.
+
+## `bijective=false`
+Given an `AbstractVector` ``[f_1, f_2, ..., f_{n}]``, returns a `Categorical` distribution,
+with probabilities given by ``l(f_1, f_2, ..., f_{n})``.
+This variant is over-parametrized, as there are `n-1` independent parameters 
+embedded in a `n` dimensional parameter space.
+For more details, see the end of the section of this [Wikipedia link](https://en.wikipedia.org/wiki/Exponential_family#Table_of_distributions)
+where it corresponds to Variant 1 and 2.
 """
-struct CategoricalLikelihood{Tl<:AbstractLink} <: AbstractLikelihood
+struct CategoricalLikelihood{Tb,Tl<:AbstractLink} <: AbstractLikelihood
     invlink::Tl
+    CategoricalLikelihood{Tb}(invlink::Tl) where {Tb,Tl} = new{Tb,Tl}(invlink)
 end
 
-CategoricalLikelihood(l=softmax) = CategoricalLikelihood(Link(l))
+function CategoricalLikelihood(l=softmax, bijective=true)
```

Comment:
Maybe make `bijective` a keyword argument? Type inference here relies on constant propagation, in general it won't be inferrable.

Reply:
Wouldn't that be the advantage of having a singleton type `Bijective` ?

Reply:
You get type inference if you use e.g.
```suggestion
function CategoricalLikelihood(l=softmax; bijective=Val(true))
```
And in case the choice is fixed (which might often be the case), then constant propagation should work. But since it can fail sometimes, it would be good to support `Val` (possibly in addition to regular booleans).

An advantage of boolean values (types) would be that it clearly indicates there are only two choices.

Reply:
And, of course, you don't have to come up with names for singleton types and also don't have to export them.

Reply:
Ok but forcing users to use `Val` is really weird IMO

Reply:
I'm used to it from SciML :smile: And you don't have to force it, you could support both.

Reply:
And AFAIK we can't multiple dispatch on keywords...
But we could have
```
CategoricalLikelihood(l=softmax, bijective::Val)
CategoricalLikelihood(l=softmax; bijective=true)
```
?

Reply:
I thought you could use something like
```suggestion
function CategoricalLikelihood(l=softmax; bijective=Val(true))
    return CategoricalLikelihood{bijective_typeparameter(bijective)}(l)
end

bijective_typeparameter(bijective::Bool) = bijective
bijective_typeparameter(::Val{T}) = T
```